### PR TITLE
docs: add PRJ-level code review quick-reference SOP (FXB-2109)

### PR DESCRIPTION
**Pull request by ryosaeba1985**
## Summary
- Creates `rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md` adapting COR-1709 for the FXB project
- §1-2: AI code review + short-form PR checklists tailored to this project's deps and commands
- §3: Actual tooling configs from `pyproject.toml` (Black, mypy, pytest, bandit)
- §4: Project-specific convention checks (railway-oriented programming, `@impure_safe`, immutable dataclasses, `Sequence[T]` params, error hierarchy)
- Registers FXB-2109 in `FXB-0000-REF-Document-Index.md`

## Test plan
- [x] `af validate` passes (94 documents, 0 issues)
- [x] Document follows SOP template per COR-1000

Closes #69

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+- [ ] H-4.1 **Pure vs IO separation**: functions that touch the filesystem are decorated with `@impure_safe` and return `IOResult[T, Error]`. Pure functions have no side effects. [P1]
+- [ ] H-4.2 **Railway-oriented programming**: pipelines use `flow()`, `bind()`, `lash()` from `returns.pipeline` / `returns.pointfree` — no bare try/except for control flow. [P1]
+- [ ] H-4.3 **Immutable data classes**: all `@dataclass` use `frozen=True`. Mutations create new instances. [P1]
+- [ ] H-4.4 **Type annotations**: function parameters accepting sequences use `Sequence[T]`, not `Tuple[T, ...]`. Return types use `List[T]` for mutable lists. [P1]
+- [ ] H-4.5 **Partial over lambda**: `functools.partial` is used to bind parameters instead of inline lambdas in pipeline chains. [P2]
+- [ ] H-4.6 **Shared types**: common types (`EntryType`, `FileBackup`, `FolderContext`) are imported from `fx_bin.shared_types`, not duplicated. [P1]
+- [ ] H-4.7 **Error hierarchy**: custom errors extend `FileOperationError` (base) or its subtypes (`ReplaceError`, `IOError`, `SecurityError`). No bare `Exception` raises. [P1]
```
[the remark this answers is not in the document]
Comment (ryosaeba1985) [reply rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md]: Good catch. Verified in fx_bin/errors.py: SecurityError extends FxBinError directly, NOT FileOperationError. Same for ValidationError, SizeError, FindError, OpenError.

Fixed in 42b4118: H-4.8 now shows the correct hierarchy - FxBinError as root, FileOperationError for file-operation errors only, and a warning that except FileOperationError does NOT catch security violations.

File: rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  poetry run bandit -r fx_bin/ && \
+  poetry run safety check && \
+  poetry run pytest
+```
+
+### 4. Project-Specific Convention Checks
+
+Reviewers must verify AI-generated code respects these FXB conventions (documented in `CLAUDE.md`):
+
+- [ ] H-4.1 **Pure vs IO separation**: functions that touch the filesystem are decorated with `@impure_safe` and return `IOResult[T, Error]`. Pure functions have no side effects. [P1]
```
[the remark this answers is not in the document]
Comment (ryosaeba1985) [reply rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md]: Fair point. Checked backup_utils.py::create_backup and organize_functional.py::get_file_date - both return IOResult directly without @impure_safe.

Fixed in 42b4118: H-4.1 now says IO functions return IOResult either via @impure_safe OR by constructing IOResult directly with explicit try/except.

File: rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+- [ ] H-4.3 **RequiresContext pattern**: IO functions that need shared context use `RequiresContext[IOResult[T, Error], ContextType]` to make dependencies explicit, not global state or closures. [P1]
+- [ ] H-4.4 **Immutable data classes**: all `@dataclass` use `frozen=True`. Mutations create new instances. [P1]
+- [ ] H-4.5 **Type annotations**: function parameters accepting sequences use `Sequence[T]`, not `Tuple[T, ...]`. Return types use `List[T]` for mutable lists. [P1]
+- [ ] H-4.6 **Partial over lambda**: `functools.partial` is used to bind parameters instead of inline lambdas in pipeline chains. [P2]
+- [ ] H-4.7 **Shared types**: common types (`EntryType`, `FileBackup`, `FolderContext`) are imported from `fx_bin.shared_types`, not duplicated. [P1]
+- [ ] H-4.8 **Error hierarchy**: custom errors extend `FileOperationError` (base) or its subtypes (`ReplaceError`, `IOError`, `SecurityError`). No bare `Exception` raises. [P1]
+- [ ] H-4.9 **CLI patterns**: `main()` entry point uses Click decorators. Registered in `pyproject.toml` under `[tool.poetry.scripts]`. [P1]
```
[the remark this answers is not in the document]
Comment (ryosaeba1985) [reply rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md]: Correct. pyproject.toml has only fx = "fx_bin.cli:main" - all subcommands use @cli.command() decorators in cli.py.

Fixed in 42b4118: H-4.9 now clarifies that only the top-level entry point goes in [tool.poetry.scripts]; subcommands use Click group decorators.

File: rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+### 2. Short-Form PR Checklist
+
+- [ ] H-2.1 Change is focused and well-described. PR title < 70 chars. [P1]
+- [ ] H-2.2 Author self-check: `poetry run pytest` passes, `poetry run black --check fx_bin/ tests/` clean, `poetry run mypy fx_bin/` clean, `poetry run flake8 fx_bin/` clean. [P0]
+- [ ] H-2.3 No hardcoded secrets, no debug code left in. `poetry run bandit -r fx_bin/` clean. [P0]
+- [ ] H-2.4 New behavior has tests; edge cases considered (including Windows paths where relevant). [P1]
+- [ ] H-2.5 No breaking changes without communication. CLI interface changes documented. [P0]
+- [ ] H-2.6 Dependencies added are necessary and safe — `poetry run safety check` passes. [P1]
+- [ ] H-2.7 Logging and error handling use `loguru` consistently. Errors use the project error hierarchy (`FileOperationError` and subtypes). [P1]
```
[the remark this answers is not in the document]
Comment (ryosaeba1985) [reply rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md]: Good catch. H-2.7 had the same narrow FileOperationError framing that H-4.8 had. Fixed: now references FxBinError root with the correct split (file-operation errors extend FileOperationError, non-file errors extend FxBinError directly).

File: rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+### 4. Project-Specific Convention Checks
+
+Reviewers must verify AI-generated code respects these FXB conventions (documented in `CLAUDE.md`):
+
+- [ ] H-4.1 **Pure vs IO separation**: IO functions return `IOResult[T, Error]` — either via `@impure_safe` (auto-catches exceptions) or by constructing `IOResult` directly with explicit try/except. Pure functions have no side effects. [P1]
+- [ ] H-4.2 **Railway-oriented programming**: pipelines use `flow()`, `bind()`, `lash()` from `returns.pipeline` / `returns.pointfree` — no bare try/except for control flow. [P1]
+- [ ] H-4.3 **RequiresContext pattern**: IO functions that need shared context use `RequiresContext[IOResult[T, Error], ContextType]` to make dependencies explicit, not global state or closures. [P1]
+- [ ] H-4.4 **Immutable data classes**: all `@dataclass` use `frozen=True`. Mutations create new instances. [P1]
```
[the remark this answers is not in the document]
Comment (ryosaeba1985) [reply rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md]: Verified. common.py has SizeEntry and FileCountEntry as plain @dataclass with __slots__ — legitimate value objects. Fixed H-4.4: now scoped to state/configuration dataclasses, with explicit carve-out for __slots__ value objects.

File: rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+  poetry run bandit -r fx_bin/ && \
+  poetry run safety check && \
+  poetry run pytest
+```
+
+### 4. Project-Specific Convention Checks
+
+Reviewers must verify AI-generated code respects these FXB conventions (documented in `CLAUDE.md`):
+
+- [ ] H-4.1 **Pure vs IO separation**: IO functions return `IOResult[T, Error]` — either via `@impure_safe` (auto-catches exceptions) or by constructing `IOResult` directly with explicit try/except. Pure functions have no side effects. [P1]
```
[the remark this answers is not in the document]
Comment (ryosaeba1985) [reply rules/FXB-2109-SOP-Code-Review-AI-Assisted-Quick-Reference.md]: Good catch. Verified: filter.py::find_files_by_extension returns List[str] and open_launcher.py::load_config raises OpenError — both legitimate imperative helpers without IOResult.

Fixed in 1c2cd02: H-4.1 now scopes the IOResult requirement to functional/railway-oriented modules (e.g., *_functional.py, backup_utils.py) and explicitly carves out imperative modules (filter.py, open_launcher.py) which may use normal returns and typed exceptions.

**Review by frankyxhl — APPROVED** [review]
